iis-workflows/iis-workflows-ingest-pmc/src/main/java/eu/dnetlib/iis/workflows/ingest/pmc/metadata/MetadataImporter.java

```diff
@@ -46,15 +85,29 @@ protected void map(AvroKey<DocumentText> key, NullWritable value, Context contex
 				saxParser.parse(new InputSource(
 						new StringReader(nlm.getText().toString())), 
 						pmcXmlHandler);
-                context.write(new AvroKey<ExtractedDocumentMetadata>(output.build()), 
-                		NullWritable.get());
+				mos.write(namedOutputMeta, 
+						new AvroKey<ExtractedDocumentMetadata>(output.build()));
         	 } catch (ParserConfigurationException e) {
-             	throw new IOException("Text extraction failed for id " + nlm.getId() + 
-             			" and text: " + nlm.getText(), e);
+        		 Map<CharSequence, CharSequence> auditSupplementaryData = new HashMap<CharSequence, CharSequence>();
+        			auditSupplementaryData.put(FAULT_TEXT, nlm.getText());
```

Comment:
This is surprising, are you sure this implicit inheritance will work? Does it work by design or by accident (the way avro currently serializes objects)? What if the fields ordering is changed in a way `BasicFaultInfo` is not the first field in record?

If it is safe to relay on this feature I guess we could agree on introducing this extendable `Fault` model. This will be both: explicitly defined and flexible approach.